va/va_enc_av1.h

```diff
@@ -0,0 +1,974 @@
+        /**
+        * \brief Tile size bytes minus1.
+        * Specify the fixed number of bytes needed to code syntax obu_size.
+        */
+        uint32_t obu_size_bytes_minus1         : 2;
+        /**
+         * \brief tx_mode supported.
+         * (tx_mode_supported & 0x01) == 1: ONLY_4X4 is supported, 0: not.
+         * (tx_mode_supported & 0x02) == 1: TX_MODE_LARGEST is supported, 0: not.
+         * (tx_mode_supported & 0x04) == 1: TX_MODE_SELECT is supported, 0: not.
+         */
+        uint32_t tx_mode_supported             : 3;
+        /**
+         * \brief Max tile num minus1.
+         * Specify the max number of tile supported by driver.
+         */
+        uint32_t max_tile_num_minus1           : 13;
+        /** \brief Reserved bits for future, must be zero. */
+        uint32_t reserved                      : 12;
+    } bits;
+    uint32_t value;
+} VAConfigAttribValEncAV1Ext2;
+
+/**
+ * \brief Packed header types specific to AV1 encoding.
+ *
+ * Types of packed headers generally used for AV1 encoding.
+ *
+ */
+typedef enum {
+    /**
+     * \brief Packed Sequence Parameter Set (SPS).
+     *
+     * The corresponding packed header data buffer shall contain the
+     * complete sequence_header_obu() syntax element.
+     *
+     */
+    VAEncPackedHeaderAV1_SPS = VAEncPackedHeaderSequence,
+    /**
+     * \brief Packed Picture Parameter Set (PPS).
+     *
+     * The corresponding packed header data buffer shall contain the
+     * complete frame_header_obu() syntax element.
+     *
+     */
+    VAEncPackedHeaderAV1_PPS = VAEncPackedHeaderPicture,
+} VAEncPackedHeaderTypeAV1;
+
+/**
+ * \brief AV1 Encoding Sequence Parameter Buffer Structure.
+ *
+ * This structure conveys sequence level parameters.
+ *
+ */
+typedef struct  _VAEncSequenceParameterBufferAV1 {
+    /** \brief AV1 profile setting.
+     *  value range [0..2].
+     */
+    uint8_t     seq_profile;
+
+    /** \brief Level Setting of current operation point.
+     *  value range [0..23].
+     */
+    uint8_t     seq_level_idx;
+
+    /** \brief Tier Setting of current operation point.
+     *  value range [0..1].
+     */
+    uint8_t     seq_tier;
+
+    uint8_t     reserved8b;
+
+    /** \brief Period between intra_only frames. */
+    uint32_t    intra_period;
+
+    /** \brief Period between I/P frames.
+     *  For hierarchical structure, this is the anchor frame distance. */
+    uint32_t    ip_period;
+
+    /* \brief RC related fields. RC modes are set with VAConfigAttribRateControl. */
+    /* For AV1, CBR implies HRD conformance and VBR implies no HRD conformance. */
+
+    /**
+     * \brief Initial bitrate set for this sequence in CBR or VBR modes.
+     *
+     * This field represents the initial bitrate value for this
+     * sequence if CBR or VBR mode is used, i.e. if the encoder
+     * pipeline was created with a #VAConfigAttribRateControl
+     * attribute set to either \ref VA_RC_CBR or \ref VA_RC_VBR.
```

Comment:
so, when its CBR, it is target bitrate, max bit rete for VBR

Reply:
yes, for VBR, the target bitrate will be generated through bits_per_second mutiplied target_percentage in #VAConfigAttribRateControl. 

Will add more comments for VBR.

Reply:
currently we set target_percentage to 1 when there are no VAConfigAttribRateContro. so if it is VBR with no VAConfigAttribRateControl. VBR will be changed to CBR at driver side.

Reply:
TBH, I think this is incorrect behavior, the behavior is not aligned with the definition.  but , other codec also defined this value. ...

Reply:
Yes, This value seems redundant and only to be used when VAConfigAttribRateControl is not set at BRC mode.